File: SretneSapice/SretneSapice.Services/Database/ForumPost.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace SretneSapice.Services.Database;

public partial class ForumPost
{
    public int PostId { get; set; }

    public int? UserId { get; set; }

    public string Title { get; set; } = null!;

    public string PostContent { get; set; } = null!;

    public DateTime? Timestamp { get; set; }

    public byte[]? Photo { get; set; }

    public byte[]? PhotoThumb { get; set; }

    public int? LikesCount { get; set; }

    public virtual ICollection<Comment> Comments { get; } = new List<Comment>();

    public virtual User? User { get; set; }

    [JsonIgnore]
    public virtual ICollection<Tag> Tags { get; } = new List<Tag>();
}

File: SretneSapice/UI/sretnesapice_admin/lib/models/forum_post.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sretnesapice_admin/models/tag.dart';
import 'package:sretnesapice_admin/models/user.dart';
import 'package:sretnesapice_admin/models/forum_post_tag.dart';
import 'package:sretnesapice_admin/models/comment.dart';

part 'forum_post.g.dart';

@JsonSerializable()
class ForumPost {
  int? postId;
  int? userId;
  String? title;
  String? postContent;
  DateTime timestamp;
  String? photo;
  List<Tag>? tags;
  List<ForumPostTag>? forumPostTags;
  User? user;
  List<Comment>? comments;

  ForumPost(this.postId, this.userId, this.title, this.postContent,
      this.timestamp, this.photo, this.tags, this.forumPostTags, this.user, this.comments);

  factory ForumPost.fromJson(Map<String, dynamic> json) =>
      _$ForumPostFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$ForumPostToJson(this);
}
